packages/flutter_tools/lib/src/device.dart

```diff
@@ -1128,7 +1132,7 @@ class DebuggingOptions {
       if (purgePersistentCache) '--purge-persistent-cache',
       if (route != null) '--route=$route',
       if (platformArgs['trace-startup'] as bool? ?? false) '--trace-startup',
-      if (enableImpeller) '--enable-impeller',
```

Comment:
don't we still need to pass --enable-impeller sometimes? I'm having a bit of a hard time wrapping my head around two independent flags 🙃

Reply:
This function creates the flags for iOS and iOS simulator. The `--enable-impeller(=true)` flag is always a no-op there after the engine flag flip. It can't override the `Info.plist` setting. When there is no 'Info.plist` setting either way, the engine flag `--enable-impeller=false` will disable Impeller on iOS and iOS sim.

After this change, when `--enable-impeller` is passed on Android, `--enable-impeller` will still be passed to the engine.

Reply:
I don't think debugging options actually needs to know, the only difference for the tooling is what shaders are generated and we've so far worked around that by just generating bot variants

Reply:
We could instead update this logic to always generate both variants for iOS and call it good?

Reply:
we already do this: https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/build_system/targets/shader_compiler.dart#L57

Reply:
Switched to an enum as discussed offline. PTAL.